targets/TARGET_Analog_Devices/TARGET_ADUCM4X50/TARGET_ADUCM4050/api/analogin_api.c

```diff
@@ -0,0 +1,212 @@
+ * DISCLAIMED. IN NO EVENT SHALL ANALOG DEVICES, INC. OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, PUNITIVE OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, DAMAGES ARISING OUT OF
+ * CLAIMS OF INTELLECTUAL PROPERTY RIGHTS INFRINGEMENT; PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ ******************************************************************************/
+
+#include "mbed_assert.h"
+#include "analogin_api.h"
+
+#if DEVICE_ANALOGIN
+
+#include "adi_adc_def.h"
+#include "pinmap.h"
+#include "PeripheralPins.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/* ADC Device number */
+#define ADC_DEV_NUM                (0u)
+
+/* Memory Required for adc driver */
+static uint32_t DeviceMemory[(ADI_ADC_MEMORY_SIZE+3)/4];
+/* Active channel */
+static uint32_t adi_pin2channel(PinName pin);
+
+/**
+ * \defgroup hal_analogin Analogin hal functions
+ * @{
+ */
+
+/** Initialize the analogin peripheral
+ *
+ * Configures the pin used by analogin.
+ * @param obj The analogin object to initialize
+ * @param pin The analogin pin name
+ */
+void analogin_init(analogin_t *obj, PinName pin)
+{
+    ADI_ADC_HANDLE hDevice;
+    bool bCalibrationDone = false;
+    bool bReady = false;
+
+    ADCName peripheral;
+    uint32_t function, channel;
+
+    peripheral = (ADCName)pinmap_peripheral(pin, &PinMap_ADC[0]);	// gives peripheral
+    MBED_ASSERT(peripheral != (ADCName)NC);
+
+    /* verify read function */
+    function = pinmap_function(pin, &PinMap_ADC[0]);
+    MBED_ASSERT(function == 1);
+
+    /* Configure PORT2_MUX registers */
+    pin_function(pin, function);
+
+    /* Configure active channel */
+    channel = adi_pin2channel(pin);
+    MBED_ASSERT(channel != 0xFFFFFFFF);
+    obj->UserBuffer.nChannels = channel;
+
+    /* Open the ADC device */
+    adi_adc_Open(ADC_DEV_NUM, DeviceMemory, sizeof(DeviceMemory), &hDevice);
+    obj->hDevice = hDevice;
+
+    /* Power up ADC */
+    adi_adc_PowerUp(hDevice, true);
+
+    /* Set ADC reference */
+    adi_adc_SetVrefSource(hDevice, ADI_ADC_VREF_SRC_INT_2_50_V);
+
+    /* Enable ADC sub system */
+    adi_adc_EnableADCSubSystem(hDevice, true);
+
+    /* Wait untilthe ADC is ready for sampling */
+    while(bReady == false) {
+        adi_adc_IsReady(hDevice, &bReady);
+    }
+
+    /* Start calibration */
+    adi_adc_StartCalibration(hDevice);
+
+    /* Wait until calibration is done */
+    while (!bCalibrationDone) {
+        adi_adc_IsCalibrationDone(hDevice, &bCalibrationDone);
+    }
+
+    /* Set the delay time */
+    adi_adc_SetDelayTime(hDevice, obj->DelayCycles);
```

Comment:
what are the values being set here and for sample cycles? It should have some default values. I can not locate setting any default value ?

Reply:
The delay and acquisition times are dependent on the sampling rate. The user application needs to configure these before acquiring samples. I don't think having a default value makes sense in this case.

Reply:
init() needs to set something. The API does not provde any function to set it, so I assume now it would use values from stack or heap, depending where AnalogIn would be allocated

Reply:
So how does the user configure the ADC if he/she wants a different sampling rate? The driver API doesn't seem to have any provisions on doing this?

Reply:
I will set some default values in the init function.

Reply:
> So how does the user configure the ADC if he/she wants a different sampling rate? The driver API doesn't seem to have any provisions on doing this?
  
Not currently supported, therefore we should choose one that should cover most use cases.

Reply:
Changes added in the latest commit.